pytorch_lightning/tuner/tuning.py

```diff
@@ -32,13 +32,22 @@ def on_trainer_init(self, auto_lr_find, auto_scale_batch_size):
         self.trainer.auto_lr_find = auto_lr_find
         self.trainer.auto_scale_batch_size = auto_scale_batch_size
 
-    def tune(self, model, train_dataloader, val_dataloaders, datamodule):
+    def setup_trainer(
+        self,
+        model: LightningModule,
+        train_dataloader: Optional[DataLoader] = None,
+        val_dataloaders: Optional[Union[DataLoader, List[DataLoader]]] = None,
+        datamodule: LightningDataModule = None,
+    ):
+        self.trainer.model_connector.copy_trainer_model_properties(model)
         # setup data, etc...
         self.trainer.train_loop.setup_fit(model, train_dataloader, val_dataloaders, datamodule)
-
         # hook
         self.trainer.data_connector.prepare_data(model)
 
+    def tune(self, model, train_dataloader, val_dataloaders, datamodule):
+        self.setup_trainer(model, train_dataloader, val_dataloaders, datamodule)
```

Comment:
also if someone has both `auto_scale_batch_size` and `auto_lr_find` enabled then also it will be called twice.

Reply:
yes but how to fix this? I don't see a good way with the current tuner design. 
We need to let the setup run every time because the user could do multiple tunings with different dataloaders etc.

Reply:
I think it's fine since it's already called multiple times when `trainer.fit` is called with new_lr/new_batch_size while tuning. you can just remove the call from `.tune`

Reply:
oh you already did :+1: